lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -172,6 +174,16 @@ struct ValueKnowledge {
       : dtype(dtype), scalarType(scalarType), kind(kind),
         optional(optionalKnowledge) {}
 
+  void print(raw_ostream &os) const {
+    os << "ValueKnowledge(";
+    if (dtype)
+      os << "dtype=" << dtype;
+    if (scalarType)
+      os << ", scalarType=" << scalarType;
+    if (optional != OptionalKnowledge::unKnown)
+      os << ", optional=" << (int)optional;
+    os << ", kind=" << (int)kind << ")";
```

Comment:
Not really necessary, per se, but it would be cleaner to print the name of the kind instead of the int value.